receiver/snmpreceiver/scraper.go

```diff
@@ -16,19 +16,46 @@ package snmpreceiver // import "github.com/open-telemetry/opentelemetry-collecto
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver/scrapererror"
 	"go.uber.org/zap"
 )
 
+var (
+	// Error messages
+	errMsgBadValueType                          = `returned metric SNMP data type for OID '%s' is not supported`
+	errMsgIndexedBadValueType                   = `returned metric SNMP data type for OID '%s' from column OID '%s' is not supported`
+	errMsgIndexedAttributesBadValueType         = `returned attribute SNMP data type for OID '%s' from column OID '%s' is not supported`
+	errMsgOIDAttributeEmptyValue                = `not creating indexed metric '%s' datapoint: %w`
+	errMsgAttributeEmptyValue                   = `metric OID attribute value is blank`
+	errMsgResourceAttributeEmptyValue           = `related resource attribute value is blank`
+	errMsgOIDResourceAttributeEmptyValue        = `not creating indexed metric '%s' or resource: %w`
+	errMsgScalarOIDProcessing                   = `problem processing scalar metric data for OID '%s': %w`
+	errMsgIndexedMetricOIDProcessing            = `problem processing indexed metric data for OID '%s' from column OID '%s': %w`
+	errMsgIndexedAttributeOIDProcessing         = `problem processing indexed attribute data for OID '%s' from column OID '%s': %w`
+	errMsgIndexedResourceAttributeOIDProcessing = `problem processing indexed resource attribute data for OID '%s' from column OID '%s': %w`
```

Comment:
```suggestion
```

Reply:
@djaglowski hmmm, so you're thinking only 1 error message? i mean this makes any given errors less immediately readable but i guess it still works since the OID is given.